automation/publish_to_maven_local_if_modified.py

```diff
@@ -103,12 +103,16 @@ def find_project_root():
     pass
 
 if contents_hash == last_contents_hash:
-    print("Contents have not changed, no need to publish")
+    print("Contents have not changed, no need to publish.")
 else:
-    print("Contents have changed, publishing")
+    print("Contents have changed, need to publish.")
     if sys.platform.startswith("win"):
+        print("Publishing latest state to a local maven repository.")
         run_cmd_checked(["gradlew.bat", "publishToMavenLocal", f"-Plocal={time.time_ns()}"], shell=True)
     else:
+        print("Cleaning up old builds from a local maven repository...")
+        run_cmd_checked(["rm -rf ~/.m2/repository/org/mozilla/components/"], shell=True)
```

Comment:
This won't work on Windows builds, I think? (I'm not sure if mhammond uses auto-publish :)

Reply:
A quick search tells me, you'd want to use something like:
```groovy
delete fileTree('$USER_HOME/.m2/repository/org/mozilla/components') {
    // filter by extension if needed.
    include '**/*.ext' 
}
```

I'm not entirely sure if the `USER_HOME` env var would work here though.

Reply:
In slack I suggested grisha is `shutil.rmtree` - I didn't notice the `~`, but `os.path.expanduser("~")` does the right thing there.

(I think that `delete fileTree` might be a powershell thing? cmd.exe is still the default shell, which wants something like `rd /S/Q ...\...`, but because we are in python, we should just stick with python)

Reply:
ie, `shutil.rmtree(os.path.expanduser("~/.m2/repository/org/mozilla/components"))` should work everywhere.

Reply:
sob - that fails if the path doesn't exist - `shutil.rmtree(..., ignore_errors=True)` - also not ideal for WSL, but probably good enough - see the a-s linked PR.